gateway/gateway.go

```diff
@@ -271,15 +260,34 @@ func (g *Gateway) Run(ctx context.Context) error {
 		return err
 	}
 
+	go func() {
```

Comment:
The cbtopology.NewPollingProvider above should be replaced but topology.NewManager returns things that I don't understand/expect when Watch is called on it so I've left it alone at the moment.